### PR TITLE
dbeaver/dbeaver#18965 use old timezone during connect

### LINKED ISSUE #18965: Change timezone from Europe/Kiev to Europe/Kyiv
### Description

timedatectl have recently changed Europe/Kiev timezone to Europe/Kyiv. As a result dbeaver doesnt work with this timezone. Please, change timezone or add new(for compatibility, some users still have Europe/Kiev timezone).
image

### DBeaver Version

Community edition 22.3.4

### Operating System

ubuntu 20.04

### Database and driver

not related

### Steps to reproduce

1. Change timezone to Europe/Kyiv
2. Open dbeaver
3. get error 
image

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Destrolaric**
Closes dbeaver/dbeaver#18965
Uses old timezone during connection, after connection succeded returns to the new one.

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreDataSource.java [hunk 1]
```diff
@@ -497,7 +499,15 @@ protected Connection openConnection(@NotNull DBRProgressMonitor monitor, @Nullab
         if (instance != null) {
             log.debug("Initiate connection to " + getServerType().getServerTypeName() + " database [" + instance.getName() + "@" + conConfig.getHostName() + "] for " + purpose);
         }
+        boolean timezoneOverridden = false;
         try {
+            // Old versions of postgres and some linux distributions, on which docker images are made, may not contain
+            // new timezone, which will lead to the error while connecting, there is no way to know before connecting
+            // so to be sure we will use the old name
+            if ("Europe/Kyiv".equals(TimeZone.getDefault().getID())) { //$NON-NLS-1$
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreDataSource.java]: Please move it to constant

Comment (serge-rider) [reply plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreDataSource.java]: What if new PG server version will have a fix for this timezone?

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreDataSource.java (hunk 1 above)
Comment (Destrolaric) [reply plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreDataSource.java]: > What if new PG server version will have a fix for this timezone?

Even if the new version will contain a new timezone, it still requires Linux distributions on which it is installed to have this timezone as well. We can't just be sure based on postgres version.

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreDataSource.java [hunk 2]
```diff
@@ -94,9 +96,11 @@ public class PostgreDataSource extends JDBCDataSource implements DBSInstanceCont
     private String activeDatabaseName;
     private PostgreServerExtension serverExtension;
     private String serverVersion;
-
     private volatile boolean hasStatistics;
 
+    private static final String LEGACY_TIMEZONE = "Europe/Kiev";
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreDataSource.java]:
```suggestion
    private static final String LEGACY_UA_TIMEZONE = "Europe/Kiev";
```

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreDataSource.java [hunk 3]
```diff
@@ -94,9 +96,11 @@ public class PostgreDataSource extends JDBCDataSource implements DBSInstanceCont
...
+    private static final String NEW_TIMEZONE = "Europe/Kyiv";
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreDataSource.java]:
```suggestion
    private static final String NEW_UA_TIMEZONE = "Europe/Kyiv";
```

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreDataSource.java (hunk 2 above)
Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreDataSource.java]: `LEGACY_TIMEZONE` is too abstract name.

**Comment by serge-rider** [comment]
This fix means that we will use old timezone always.
What happens in new PostgreSQL server version? Does it understand both old and new timezone?

**Comment by MashaKorax** [comment]
verified

**Pull request merged**